app/src/main/java/de/hhn/aib/labsw/blackmirror/controller/API/Websockets/MirrorApiWebsockets.kt

```diff
@@ -76,7 +83,7 @@ class MirrorApiWebsockets : WebSocketListener(),MirrorApi {
      * @param listener listener to remove, not null
      */
     override fun unsubscribe(topic: String, listener: ApiListener) {
-        val listenerList: MutableList<ApiListener>? = listeners[topic]
+        val listenerList  = listeners[topic]
         listenerList?.remove(listener)
```

Comment:
this should not be implemented with the safe call operator `?.` because `listenersList` should never be `null` here
instead check if it is null in which case it should throw a NPE

Reply:
the list can be null if the topic given as argument has never been subscribed to by any listener